File: front/src/domain/use-cases/auth/GetCurrentUserUseCase.ts
import { AuthSession } from '../../entities/AuthSession';
import { IAuthRepository } from '../../repositories/IAuthRepository';

export class GetCurrentUserUseCase {
  constructor(private authRepository: IAuthRepository) {}

  async execute(): Promise<AuthSession | null> {
    return await this.authRepository.getCurrentSession();
  }
}

File: Nueva carpeta/hospital_movil/lib/features/auth/domain/usecases/get_current_user_usecase.dart
import '../../../../core/usecase/usecase.dart';
import '../entities/user.dart';
import '../repositories/auth_repository.dart';

class GetCurrentUserUseCase implements UseCase<User?, NoParams> {
  final AuthRepository repository;

  GetCurrentUserUseCase(this.repository);

  @override
  Future<Result<User?>> call(NoParams params) async {
    return await repository.getCurrentUser();
  }
}
